Sources/LanguageServerProtocol/SupportTypes/FileEvent.swift

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// An event describing a file change.
+public struct FileEvent: Codable, Hashable {
+  public var uri: DocumentURI
+  public var type: FileChangeType
+
+  public init(uri: DocumentURI, type: FileChangeType) {
+    self.uri = uri
+    self.type = type
+  }
+}
+/// The type of file event.
+///
+/// In LSP, this is an integer, so we don't use a closed set.
+public struct FileChangeType: RawRepresentable, Codable, Hashable {
```

Comment:
Worth adding a test to ensure this is encoded correctly since it needs to use the raw value directly.

Reply:
Done